### ISSUE TITLE
machinekit.io needs link to docs!

### ISSUE THREAD

**Issue by zultron** [machinekit/machinekit.github.io#46]
There's no link to https://github.com/machinekit/machinekit-docs on the machinekit.io documentation page.

**Comment by RunningLight** [comment]
Agreed. I've been reluctant to direct folks to this site until we have a process in place to render it. But, with a little explanation so they don't expect to find a Wikipedia, it will be what they help make it, agreed.

I'm off celebrating a family event today, but tomorrow morning plan to clean up my proposed changes to machinekit.GitHub.io based on the comments I got back from my earlier email and make it a PR. Will add your proposed link as a 2nd PR.

**Comment by luminize** [comment]
> On 20 Jun 2015, at 17:30, Kent Reed <notifications@github.com> wrote:
> 
> Agreed. I've been reluctant to direct folks to this site until we have a process in place to render it. But, with a little explanation so they don't expect to find a Wikipedia, it will be what they help make it, agreed.

With all respect guys, apologies for being blunt. But the main reason to take the docs out of the code base was to make it easy to edit and add to.

Everything regarding docs and adding/editing, including low hurdle git-noob friendly workflow just via the GitHub portal is there.

And now we're discussing about rendering? As if that will make everybody read the docs?

No offence, but this recent rendering discussion just doesn't make sense to me.

I don't get it.
Just make one pointer to the docs repo, and force users to go there. One place for instructions, no fragmentation of info.

Bas

**Comment by zultron** [comment]
Whoa!  What happened right there?

@RunningLight, the docs are rendered right in GitHub.  Click the link in the original post above, or try https://github.com/machinekit/machinekit-docs/blob/master/machinekit-documentation/documents-index.asciidoc.

@luminize, it frustrates me, too, that nobody knows about this good, hard work.  I think if this link existed, folks could begin to find the docs.

Once we can find the docs, anything that bugs us about them can be fixed.  ;)

**Comment by luminize** [comment]
> On 20 Jun 2015, at 18:19, John Morris <notifications@github.com> wrote:
> 
> Whoa! What happened right there?
> 
Scope creep, this borders on an off line thread about documentation, rendering and google search ability.
> @RunningLight, the docs are rendered right in GitHub. Click the link in the original post above, or try https://github.com/machinekit/machinekit-docs/blob/master/machinekit-documentation/documents-index.asciidoc.
> 
> @luminize, it frustrates me, too, that nobody knows about this good, hard work. I think if this link existed, folks could begin to find the docs.
> 
I have no need for shoulder pads. I also don't intend to pick on Anybody. The docs are not my baby.

But I fully disagree that nobody knows about where to find the docs.

If people want to maintain instructions in 2 locations, fragmenting info, confusing users.... fine.... But not me.

Keep it simple, force users to go to one place, one place only.

As long as docs are not maintained, and expanded on, the list will be the place to answer questions 10 times instead of documenting the info once. Especially with new stuff.

So even pointing to the docs repo will not change the content of the repo. So as long as new stuff is not documented, the list is the only place to go.

I realize this is not about the issue of the missing link. Apologies for that.

-end of rant-

**Comment by zultron** [comment]
On 06/20/2015 11:54 AM, Bas de Bruijn wrote:
>
>
>  > On 20 Jun 2015, at 18:19, John Morris <notifications@github.com> wrote:
>  >
>  > Whoa! What happened right there?
>  >
> Scope creep, this borders on an off line thread about documentation,
> rendering and google search ability.

FWIW, I did find what I needed by knowing the search terms 'machinekit 
"code notes"' for Google.

>  > @RunningLight, the docs are rendered right in GitHub. Click the link
> in the original post above, or try
> https://github.com/machinekit/machinekit-docs/blob/master/machinekit-documentation/documents-index.asciidoc.
>  >
>  > @luminize, it frustrates me, too, that nobody knows about this good,
> hard work. I think if this link existed, folks could begin to find the docs.
>  >
> I have no need for shoulder pads. I also don't intend to pick on
> Anybody. The docs are not my baby.
>
> But I fully disagree that nobody knows about where to find the docs.
>
> If people want to maintain instructions in 2 locations, fragmenting
> info, confusing users.... fine.... But not me.
>
> Keep it simple, force users to go to one place, one place only.
>
> As long as docs are not maintained, and expanded on, the list will be
> the place to answer questions 10 times instead of documenting the info
> once. Especially with new stuff.
>
> So even pointing to the docs repo will not change the content of the
> repo. So as long as new stuff is not documented, the list is the only
> place to go.
>
> I realize this is not about the issue of the missing link. Apologies for
> that.

Before the 'end of rant' marker ;), I'd like to chime in on this, even 
though it should be in another issue or on the list.  You're absolutely 
right that docs should be regularly updated, and this relates to my 
concern that we're not enforcing the C4 spec saying that APIs should be 
accompanied by documentation.  This should be something that Maintainers 
look for as part of the minimal review before merge, and I've certainly 
been guilty of ignoring it.  So that problem does have a social as well 
as a technical component that we must raise awareness of.

> -end of rant-

:)

Back to the issue at hand, it's true that I didn't know where to find 
the docs, and when I went to where I expected I could find out 
(http://www.machinekit.io/docs/home/), no link there.

And to your point about "one place" for docs, arguably, two of the links 
under "Related material" should be off that page and inside the docs 
(and the third one on the blog).

	John

**Comment by RunningLight** [comment]
I'm the one who owes an apology, full stop. 

If you're feeling charitable (and no reason you should be at this point), chalk it up to old-timers disease. I somehow crafted a "convenient" working environment where I end up seeing raw .asciidoc files. Starting with a clean slate---"zip zap zowie and swoosh" [1], I see rendered pages. 

Forget what I said about the work I did before. I'm happy to see everything rolled into machinekit-docs instead of tweaking pages on machinekit.io. No wonder @luminize is getting frustrated. He has every right to be.

[1] the best comment I ever came across in Digital Equipment Corp. source for RSX-11 ca 1971.

**Comment by RunningLight** [comment]
PS - some of my earlier comments about rendering .asciidoc still come into play if you consider publishing machinekit-docs as a static website via GitHub Pages the same way we publish machinekit.github.io, but that's an issue which can be taken up separately.

**Comment by RunningLight** [comment]
I still think such a link from machinekit.github.io to machinekit-docs needs some explanatory text and I'll propose something tonight if no one else has posted a PR in the meantime (sorry, it's just too nice outside right now to keep typing).

I have composed a matrix showing what documentation exists on machinekit.io now and what I had proposed recently as a rework of it; now I'm adding a 3rd column to the matrix showing how I imagine the contents might be moved into specific machinekit-docs locations in keeping with a "Keep it simple, force users to go to one place, one place only" strategy. I'll post it tonight as a new issue.
